From 71c908a825d6bdb217a9a5aef984c49247e95c45 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Fri, 29 Jan 2021 19:19:48 +0800
Subject: [PATCH] drm/rockchip: vop2: Fix wait frame start condition check

When system boot with boot logo on, drm_crtc_atomit_enable
won't be called, so vcstate->vdisplay can't be initialized.

So we get vdisplay/vtotatal by crtc.state->adjusted_mode.

Fixes: bd11b0fed848 ("drm/rockchip: vop2: wait for frame start by vcnt")

Change-Id: I39a1d62095347fe6e5d3c48ff7b1e08a20366f9c
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index e249e4547bcd..923e25242ddc 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -757,7 +757,7 @@ static inline void vop2_cfg_done(struct drm_crtc *crtc)
 {
 	struct vop2_video_port *vp = to_vop2_video_port(crtc);
 	struct vop2_video_port *done_vp;
-	struct rockchip_crtc_state *vcstate;
+	struct drm_display_mode *adjusted_mode;
 	struct vop2 *vop2 = vp->vop2;
 	uint32_t done_bits;
 	uint32_t vp_id;
@@ -787,10 +787,10 @@ static inline void vop2_cfg_done(struct drm_crtc *crtc)
 		/* no need to wait for same vp */
 		if (vp_id != vp->id) {
 			done_vp = &vop2->vps[vp_id];
-			vcstate = to_rockchip_crtc_state(done_vp->crtc.state);
+			adjusted_mode = &done_vp->crtc.state->adjusted_mode;
 			vcnt = vop2_read_vcnt(done_vp);
 			/* if close to the last 1/4 frame, wait to next frame */
-			if (vcnt > (vcstate->vdisplay * 3 >> 2)) {
+			if (vcnt > (adjusted_mode->crtc_vtotal * 3 >> 2)) {
 				vop2_wait_for_fs_by_vcnt(done_vp, vcnt);
 				done_bits = 0;
 			}
-- 
2.35.3

